interpreter/interpretable.go

```diff
@@ -748,3 +726,345 @@ func (fold *evalExhaustiveFold) Eval(ctx Activation) ref.Val {
+	for _, attr := range e.attrs {
+		v, found := e.resolver.Resolve(vars, attr)
+		if found {
+			return e.adapter.NativeToValue(v)
+		}
+	}
+	return types.NewErr("no such attribute")
+}
+
+func (e *evalOneofAttr) addField(id int64, name types.String) attrInst {
+	// When a field is added to a oneof attribute, it should be appended to the name of the
+	// top-most variable as a possible namespaced identifier, and then as a possible field
+	// on all less-specified attributes.
+	modAttrs := make([]Attribute, 0, len(e.attrs))
+	pe := newExprPathElem(id, name)
+	for _, attr := range e.attrs {
+		if len(attr.Path()) != 0 {
+			continue
+		}
+		v := attr.Variable()
+		modName := fmt.Sprintf("%s.%s", v.Name(), name)
+		modAttrs = append(modAttrs, newExprVarAttribute(id, modName))
+	}
+	for _, attr := range e.attrs {
+		modAttrs = append(modAttrs, attr.Select(pe))
+	}
+	return &evalOneofAttr{
+		id:       e.id,
+		adapter:  e.adapter,
+		attrs:    modAttrs,
+		resolver: e.resolver,
+	}
+}
+
+func (e *evalOneofAttr) addIndex(pe *PathElem) attrInst {
+	// adding an index is not the same as adding a field since the index cannot possibly be an
+	// identifier, so the selection behavior differs here.
+	modAttrs := make([]Attribute, len(e.attrs), len(e.attrs))
+	for i, attr := range e.attrs {
+		modAttrs[i] = attr.Select(pe)
+	}
+	return &evalOneofAttr{
+		id:       e.id,
+		adapter:  e.adapter,
+		attrs:    modAttrs,
+		resolver: e.resolver,
+	}
+}
+
+func (e *evalOneofAttr) getAdapter() ref.TypeAdapter {
+	return e.adapter
+}
+
+func (e *evalOneofAttr) getResolver() Resolver {
+	return e.resolver
+}
+
+func (e *evalOneofAttr) getAttrs() []Attribute {
+	return e.attrs
+}
+
+// evalRelAttr specifies relative attribute selection against a computed value.
+type evalRelAttr struct {
+	id       int64
+	adapter  ref.TypeAdapter
+	resolver Resolver
+	op       Interpretable
+	attr     Attribute
+}
+
+// ID implements the Interpretable interface method.
+func (e *evalRelAttr) ID() int64 {
+	return e.id
+}
+
+// Eval implements the Interpretable interface method.
+func (e *evalRelAttr) Eval(vars Activation) ref.Val {
+	obj := e.op.Eval(vars)
+	val := e.resolver.ResolveRelative(obj, vars, e.attr)
+	return e.adapter.NativeToValue(val)
+}
+
+func (e *evalRelAttr) addField(id int64, name types.String) attrInst {
+	// treat field selection as identical to index operations.
+	return e.addIndex(newExprPathElem(id, name))
+}
+
+func (e *evalRelAttr) addIndex(pe *PathElem) attrInst {
+	return &evalRelAttr{
+		id:       e.id,
+		adapter:  e.adapter,
+		attr:     e.attr.Select(pe),
+		op:       e.op,
+		resolver: e.resolver,
+	}
+}
+
+func (e *evalRelAttr) getAdapter() ref.TypeAdapter {
+	return e.adapter
+}
+
+func (e *evalRelAttr) getResolver() Resolver {
+	return e.resolver
+}
+
+func (e *evalRelAttr) getAttrs() []Attribute {
+	return []Attribute{e.attr}
+}
+
+// evalConstEq describe the case where an attribute is compared against a constant value.
```

Comment:
If `evalConstEq` means "the set of objects that conform to  `evalConstEq`", then replace "the" with "an" in "When the evalConstEq is produced", otherwise add an S to `describe**s**`. Same in evalConstNe